File: labs/7/BookReviewsPartDeux/app/src/main/java/me/henryfbp/sqlitelabpartdeux/ListAdapterBook.java
package me.henryfbp.sqlitelabpartdeux;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.RatingBar;
import android.widget.TextView;

import java.util.List;

public class ListAdapterBook extends ArrayAdapter<Book> {

    private List<Book> items;

    public ListAdapterBook(Context context, int resource, List<Book> items) {
        super(context, resource, items);
        this.items = items;
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        View v = convertView;

        if (v == null) {

            LayoutInflater vi;
            vi = LayoutInflater.from(getContext());
            v = vi.inflate(R.layout.itemlistrow, parent);
        }

        Book p = getItem(position);

        if (p != null) {

            TextView tt = v.findViewById(R.id._id);
            TextView tt1 = v.findViewById(R.id.title);
            TextView tt3 = v.findViewById(R.id.author);
            RatingBar rb = v.findViewById(R.id.rating);

            if (tt != null) {
                tt.setText(p.getId().toString());
            }

            if (tt1 != null) {
                tt1.setText(p.getTitle());
            }

            if (tt3 != null) {
                tt3.setText(p.getAuthor());
            }

            if (rb != null) {
                rb.setRating(p.getRating());
            }
        }
        return v;
    }
}

File: labs/8/BookReviewsPartTrois/app/src/main/java/me/henryfbp/sqlitelabparttrois/ListAdapterBook.kt
package me.henryfbp.sqlitelabparttrois

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.RatingBar
import android.widget.TextView

class ListAdapterBook(context: Context, resource: Int, items: List<Book>) : ArrayAdapter<Book>(context, resource, items) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        var v = convertView

        if (v == null) {

            val vi: LayoutInflater = LayoutInflater.from(context)
            v = vi.inflate(R.layout.item_single_book, null)
        }

        val p = getItem(position)

        if (p != null) {

            val tt = v!!.findViewById<TextView>(R.id._id)
            val tt1 = v.findViewById<TextView>(R.id.title)
            val tt3 = v.findViewById<TextView>(R.id.author)
            val rb = v.findViewById<RatingBar>(R.id.rating)

            if (tt != null) {
                tt.text = p.id!!.toString()
            }

            if (tt1 != null) {
                tt1.text = p.title
            }

            if (tt3 != null) {
                tt3.text = p.author
            }

            if (rb != null) {
                rb.rating = p.rating
            }
        }

        return v!!
    }
}
